2018/003-offline-chart-repository.md

```diff
@@ -0,0 +1,441 @@
+  * `/usr/share/suse-kubernetes-charts/console-0.9.9.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.0.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.0.2.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.1.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.0.2.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.10.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.6.11.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.7.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.8.0.tgz`
+
+The same thing will be done by the other teams at SUSE who want to ship containerized
+workloads managed by helm.
+
+#### Building the offline chart repository image
+
+The `registry.suse.com/suse/kubernetes-charts` will be built inside of the
+Open Build Service (obviously inside of IBS).
+
+The build service project will consists of:
+
+  * RPM packages of helm charts.
+  * RPM defining patterns of helm charts.
+  * Definition of the `registry.suse.com/suse/kubernetes-charts` container image.
+
+The next sections will provide details about these resources.
+
+##### RPM packages definitions
+
+Each helm chart should have its own RPM package. For example we would have a
+`cf-chart`, `console-chart`, `uaa-chart`, …
+
+We must be able to ship multiple versions of the same helm chart (eg: `cf-2.0.2.tgz` and `cf-2.10.1.tgz`). To achieve that each RPM will install one `.tgz` file per version of the chart.
+
+For example, the build service package for the `cf-chart` RPM would consist of the following files:
+
+  * `cf-chart.spec`
+  * `cf-chart.changelog`
+  * `cf-2.0.2.tgz`
+  * `cf-2.10.1.tgz`
+
+Once installed the RPM would provide the following files:
+
+  * `/usr/share/suse/kubernetes-charts/cf-2.0.2.tgz`
+  * `/usr/share/suse/kubernetes-charts/cf-2.10.1.tgz`
+
+In the future each chart RPM should also provide the `.tgz.prov` provenance files. However this RFC doesn't want to cover the problem of creating these signed files (currently they are missing also from *”kubernetes-charts.suse.com”*).
+
+Each team would retain ownership of these packages. That would allow them to move at
+their own pace and not rely on other teams to process their submission requests.
+
+##### RPM pattern definitions
+
+Each team will have its own RPM defining a pattern that requires all the single
+helm charts that are related with it.
+
+For example the Cloud Application Platform team could have a `cap-charts`
+pattern that requires the different packages providing the cloud foundry charts (eg: `cf-chart`, `uaa-chart`, ...).
+
+Each team would retain ownership of their pattern packages.
+That would allow them to move at their own pace and not rely on other teams to
+process their submission requests.
+
+##### Container image definition
+
+The OBS project would include also the definition of the `registry.suse.com/suse/kubernetes-charts` container image.
+
+This image would take the ChartMuseum image maintained by the SUSE CaaS Platform
```

Comment:
Why do we need 2 CM builds, rather than have one kiwi project that can pull in multiple dependencies?